presto-main/src/main/java/com/facebook/presto/execution/ClusterSizeMonitor.java

```diff
@@ -153,28 +187,74 @@ public synchronized void verifyInitialMinimumWorkersRequirement()
         // remove future if finished (e.g., canceled, timed out)
         future.addListener(() -> {
             timeoutTask.cancel(true);
-            removeFuture(future);
+            removeWorkerFuture(future);
         }, executor);
 
         return future;
     }
 
-    private synchronized void removeFuture(SettableFuture<?> future)
+    public synchronized ListenableFuture<?> waitForMinimumCoordinators()
     {
-        futures.remove(future);
+        if (currentCoordinatorCount >= coordinatorMinCount) {
+            return immediateFuture(null);
+        }
+
+        SettableFuture<?> future = SettableFuture.create();
+        coordinatorSizeFutures.add(future);
+
+        // if future does not finish in wait period, complete with an exception
+        ScheduledFuture<?> timeoutTask = executor.schedule(
+                () -> {
+                    synchronized (this) {
+                        future.setException(new PrestoException(
+                                GENERIC_INSUFFICIENT_RESOURCES,
+                                format("Insufficient active coordinator nodes. Waited %s for at least %s coordinators, but only %s coordinators are active", executionMaxWait, 2, currentCoordinatorCount)));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format int: currentCoordinatorCount. This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 211. Consider formatting this data with &#39;%d&#39; instead.